platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -316,10 +317,10 @@ public static synchronized void stop() {
 
     DEFAULT_TAGS.clear();
     METER_REGISTRY.forEachMeter(METER_REGISTRY::remove);
-    METER_REGISTRY.getRegistries().forEach(e -> {
-      e.clear();
-      METER_REGISTRY.remove(e);
-    });
+    METER_REGISTRY.getRegistries().forEach(MeterRegistry::clear);
+    Set<MeterRegistry> registries = new HashSet<>(METER_REGISTRY.getRegistries());
+    registries.forEach(METER_REGISTRY::remove);
+    registries.clear();
```

Comment:
I'm confused. The clear is on the new HashSet object you are creating. So the underlying `METER_REGISTRY.getRegistries()` still has the registries right?

Reply:
Though we got rid of the problem, if the intention is to clear those registries I think that wouldn't happen

Reply:
@avinashkolluru  `registries.forEach(METER_REGISTRY::remove)`  will remove it from METER_REGISTRY.remove()
<img width="1661" alt="Screenshot 2020-08-26 at 11 31 46 PM" src="https://user-images.githubusercontent.com/64777291/91339634-632ab080-e7f4-11ea-9a5c-910a3b8b2ffc.png">

Reply:
Dumb. Sorry.

Reply:
@samarth-gupta-traceable can't you use iterator and do `itr.remove()` instead?